File: 2016. Maximum Difference Between Increasing Elements/maximumDifference.cpp
#include <vector>
#include <iostream>

using namespace std;

class Solution {
    public:
        int maximumDifference(vector<int>& nums) {
            int minNum = nums[0];
            int res = -1;
    
            for(int i=1 ; i<nums.size(); i++){
                if(nums[i] > minNum){
                    res = max(res, nums[i]-minNum);
                }
                minNum = min(minNum, nums[i]);
            }
    
            return res;
        }
    };

File: 2016. Maximum Difference Between Increasing Elements/maximumDifference.js
/**
 * @param {number[]} nums
 * @return {number}
 */

//SOLUTION #1
// T.C => O(n^2)
var maximumDifference = function (nums) {
    let res = -1;

    for (let i = 0; i < nums.length; i++) {

        for (let j = i + 1; j < nums.length; j++) {

            res = Math.max(res, nums[j] - nums[i])
        }
    }

    return res ? res : -1;
};

//SOLUTION #2
// T.C => O(n)
var maximumDifference = function (nums) {
    let min = nums[0];
    let res = -1;


    for (let i = 1; i < nums.length; i++) {
        if (nums[i] > min) {
            res = Math.max(res, nums[i] - min)
        }
        min = Math.min(min, nums[i])
    }

    return res
};